android/sdl_android/src/main/java/com/smartdevicelink/transport/SdlBroadcastReceiver.java

```diff
@@ -232,6 +246,16 @@ private static void startRouterService(Context context, ComponentName componentN
             serviceIntent.putExtra(TransportConstants.CONFIRMED_SDL_DEVICE, confirmedDevice);
         }
 
+        if (vehicleType == null) {
+            Hashtable<String, Object> store = AndroidTools.getVehicleTypeFromPrefs(context, device.getAddress());
```

Comment:
There is a second part to this, that is once we get the RAI response on the first connection, and we haven't checked system info, we are going to save the vehicle info to shared prefs. We can do this in the `lifecycleManager` .  We can get the Bluetooth mac address from `SdlProtocolBase`.

Method to add to `SdlProtocolBase`

```
    public String getBluetoothMacAddress() {
        for(TransportRecord record : activeTransports.values()) {
            if(record.getType() == TransportType.BLUETOOTH){
                return record.getAddress();
            }
        }

        return null;
    }
```

Changes to `BaseSdlSession`. 

```
    public String getBluetoothMacAddress() {
        if (this.sdlProtocol != null) {
            return this.sdlProtocol.getBluetoothMacAddress();
        } else {
            return null;
        }
    }
```

Changes to `BaseLifeCycleManager`

Then we can add this part where we check for SystemInfo once we get the RAI response in `OnRPCListener`

```
String address = session.getBluetoothMacAddress();
if (address != null && !address.isEmpty()) {
    saveVehicleType(address, vehicleType);
}
```
to make the section look like:
```
                        if (!didCheckSystemInfo && lifecycleListener != null) {
                            didCheckSystemInfo = true;
                            VehicleType vehicleType = raiResponse.getVehicleType();
                            String systemSoftwareVersion = raiResponse.getSystemSoftwareVersion();
                            if (vehicleType != null || systemSoftwareVersion != null) {
                                String address = session.getBluetoothMacAddress();
                                if (address != null && !address.isEmpty()) {
                                    saveVehicleType(address, vehicleType);
                                }
                                SystemInfo systemInfo = new SystemInfo(vehicleType, systemSoftwareVersion, null);
                                boolean validSystemInfo = lifecycleListener.onSystemInfoReceived(systemInfo);
                                if (!validSystemInfo) {
                                    DebugTool.logWarning(TAG, "Disconnecting from head unit, the system info was not accepted.");
                                    UnregisterAppInterface msg = new UnregisterAppInterface();
                                    msg.setCorrelationID(UNREGISTER_APP_INTERFACE_CORRELATION_ID);
                                    sendRPCMessagePrivate(msg, true);
                                    clean();
                                    return;
                                }
                            }
                        }
```

Then add to `BaseLifecycleManager` :
```
abstract void saveVehicleType(String address, VehicleType type);
```
There is a section with other abstract methods to add this to.

Then we can implement this method in `LifecycleManager` to save the vehicle type to prefs 

```
    @Override
    void saveVehicleType(String address, VehicleType type){
        AndroidTools.saveVehicleType(contextWeakReference.get(), type, address);
    }
```